packages/vite/src/node/plugins/define.ts

```diff
@@ -49,16 +57,20 @@ export function definePlugin(config: ResolvedConfig): Plugin {
   const importMetaKeys: Record<string, string> = {}
   const importMetaFallbackKeys: Record<string, string> = {}
   if (isBuild) {
-    env.SSR = !!config.build.ssr
-
+    const env: Record<string, any> = {
+      ...config.env,
+      SSR: !!config.build.ssr,
+    }
     // set here to allow override with config.define
     importMetaKeys['import.meta.hot'] = `undefined`
     for (const key in env) {
       importMetaKeys[`import.meta.env.${key}`] = JSON.stringify(env[key])
     }
     Object.assign(importMetaFallbackKeys, {
       'import.meta.env.': `({}).`,
-      'import.meta.env': JSON.stringify(env),
+      'import.meta.env': JSON.stringify({ ...env, ...userDefineEnv })
```

Comment:
Notice that `import.meta.env.SSR` will be replaced by user define values.

Reply:
I think that might be fine for now as in dev (import analysis), it's allowed to overwrite `import.meta.env.SSR` too 🤔 

https://github.com/vitejs/vite/blob/3551f754a6b141cd843dc061f6b19ea80f5c6497/packages/vite/src/node/plugins/importAnalysis.ts#L636-L652

this code injects the `import.meta.env` code for each file. Also I just realized we could probably cache this code.